### PR TITLE
Add RequireLastPushApproval field to UpdateBranchProtection

### PR DESCRIPTION
**Pull request by liaodaniel**
Fixes https://github.com/google/go-github/issues/2628

Signed-off-by: Daniel Liao <10663736+liaodaniel@users.noreply.github.com>

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -1064,6 +1064,8 @@ type PullRequestReviewsEnforcementRequest struct {
 	// RequiredApprovingReviewCount specifies the number of approvals required before the pull request can be merged.
 	// Valid values are 1-6.
 	RequiredApprovingReviewCount int `json:"required_approving_review_count"`
+	// RequireLastPushApproval specifies whether the last pusher to a pull request branch can approve it.
+	RequireLastPushApproval bool `json:"require_last_push_approval"`
```
Comment (gmlewis) [inline github/repos.go:1068]:
```suggestion
	RequireLastPushApproval *bool `json:"require_last_push_approval,omitempty"`
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @liaodaniel!
This field is optional, so we need to treat it a bit differently.

File: github/repos.go (hunk 1 above)
Comment (liaodaniel) [reply github/repos.go]: Updated with the suggestion

**Review by gmlewis — APPROVED** [review]
Thank you, @liaodaniel !
LGTM.
Merging.

**Pull request merged**